harness/determined/pytorch/samplers.py

```diff
@@ -0,0 +1,361 @@
+"""
+Guidelines for Reproducible Datasets.
+
+.. note::
+
+   If you are using a MapDataset and not an IterableDataset, make your life easy and use the
+   context.TrainingDataLoader() and context.ValidationDataLoader(), which are a drop-in replacement
+   for pytorch's built-in DataLoader, and Determined will handle all of these details.
+
+Achieving a reproducible dataset that is able to pause and continue (sometimes called "incremental
+training") is easy if you follow a few rules.
+
+- Even if you are going to ultimate return an IterableDataset, it is best to use PyTorch's Sampler
+  class as the basis for choosing the order of which records to read.  Operations on Samplers are
+  quick and cheap, while operations on data afterwards are expensive.  For more details, see the
+  discussion at TODO:ref_yogadl_readme.  If you don't have a custom sampler, start with a simple
+  one:
```

Comment:
"choosing the order of records"

Reply:
"Operations on Samplers are
  quick and cheap, while operations on data afterwards are expensive.  For more details, see the
  discussion at TODO:ref_yogadl_readme. " is a bit confusing. Can we add a one line sentence that elaborate a bit more. Maybe though the TODO will help though

Reply:
it will point to yogadl.readthedocs.io, do you think that would be helpful?  (that's the README you helped edit)